### PR TITLE
[#1492] Add testing-str to analysis context via built-in hook

### PR DESCRIPTION
**Pull request by borkdude**
Please answer the following questions and leave the below in as part of your PR.

- [x] I have read the [developer documentation](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md).

- [x] This PR correponds to an [issue with a clear problem statement](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md#start-with-an-issue-before-writing-code).

Issue: #1492 

- [x] This PR contains a [test](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md#tests) to prevent against future regressions

### CODE REVIEW

**Comment by borkdude** [comment]
@ericdallo I added the testing-str to the analysis only of the testing var usage. Can you check if this works for your use case?

File: src/clj_kondo/impl/analyzer/test.clj [hunk 1]
```diff
@@ -40,3 +40,10 @@
                                                                     is-expr
                                                                     args)))]
     (common/analyze-expression** ctx new-node)))
+
+(defn testing-hook [{:keys [node]}]
+  (let [[testing testing-str & children] (:children node)
+        new-node (assoc node :children
+                        (list* (assoc-in testing [:context :clojure.test :testing-str] (utils/sexpr testing-str))
```
Comment (ericdallo) [inline src/clj_kondo/impl/analyzer/test.clj:47]: Should we consider `cljs.test` as well?

Comment (borkdude) [reply src/clj_kondo/impl/analyzer/test.clj:47]: Do you mean the keyword? I think we can use the same keyword for both dialects?

**Comment by ericdallo** [comment]
@borkdude I tested with clojure-lsp and now it seems there is a extra var-usage on the same position but with the context, shouldn't we add the context to the existing var-usage instead of adding a new one?
The old:
```clojure
{:end-row 22,
    :name-end-col 11,
    :name-end-row 20,
    :name-row 20,
    :name testing,
    :filename
    "/home/greg/dev/clojure-lsp/test/clojure_lsp/shared_test.clj",
    :from clojure-lsp.shared-test,
    :macro true,
    :col 3,
    :name-col 4,
    :context {},
    :end-col 92,
    :arity 2,
    :varargs-min-arity 1,
    :bucket :var-usages,
    :row 20,
    :to clojure.test}
```
and the new:
```clojure
{:end-row 22,
    :name-end-col 11,
    :name-end-row 20,
    :name-row 20,
    :name testing,
    :filename
    "/home/greg/dev/clojure-lsp/test/clojure_lsp/shared_test.clj",
    :from clojure-lsp.shared-test,
    :macro true,
    :col 3,
    :name-col 4,
    :context
    {:clojure.test
     {:testing-str "should decode special characters in file URI"}},
    :from-var uri->filename,
    :end-col 92,
    :arity 2,
    :varargs-min-arity 1,
    :bucket :var-usages,
    :row 20,
    :to clojure.test}
```

**Comment by borkdude** [comment]
Good point! Will fix

**Comment by borkdude** [comment]
@ericdallo Can you test with the new commit? Still need to make a test for it.

**Comment by ericdallo** [comment]
@borkdude now we do have only one testing var-usage analysis indeed but with empty context 😅

**Comment by ericdallo** [comment]
Actually, there are another var-usage analysis:
 ```clojure
 (ns clojure-lsp.refactor.bla-test
  (:require
    [clojure.test :refer [deftest testing]]))

(deftest foo
  (testing "bar 1"))
 ```
 var-usages
 ```json
 [
  {
    "end-row": 3,
    "name-end-col": 34,
    "name-end-row": 3,
    "name-row": 3,
    "name": "deftest",
    "filename": "../clojure-lsp/test/clojure_lsp/refactor/bla_test.clj",
    "from": "clojure-lsp.refactor.bla-test",
    "macro": true,
    "col": 27,
    "name-col": 27,
    "context": {},
    "end-col": 34,
    "varargs-min-arity": 1,
    "refer": true,
    "row": 3,
    "to": "clojure.test"
  },
  {
    "end-row": 3,
    "name-end-col": 42,
    "name-end-row": 3,
    "name-row": 3,
    "name": "testing",
    "filename": "../clojure-lsp/test/clojure_lsp/refactor/bla_test.clj",
    "from": "clojure-lsp.refactor.bla-test",
    "macro": true,
    "col": 35,
    "name-col": 35,
    "context": {},
    "end-col": 42,
    "varargs-min-arity": 1,
    "refer": true,
    "row": 3,
    "to": "clojure.test"
  },
  {
    "end-row": 6,
    "name-end-col": 11,
    "name-end-row": 6,
    "name-row": 6,
    "name": "testing",
    "filename": "../clojure-lsp/test/clojure_lsp/refactor/bla_test.clj",
    "from": "clojure-lsp.refactor.bla-test",
    "macro": true,
    "col": 3,
    "name-col": 4,
    "context": {
      "clojure.test": {
        "testing-str": "bar 1"
      }
    },
    "from-var": "foo",
    "end-col": 20,
    "arity": 1,
    "varargs-min-arity": 1,
    "row": 6,
    "to": "clojure.test"
  },
  {
    "end-row": 6,
    "name-end-col": 9,
    "name-end-row": 5,
    "name-row": 5,
    "name": "deftest",
    "filename": "../clojure-lsp/test/clojure_lsp/refactor/bla_test.clj",
    "from": "clojure-lsp.refactor.bla-test",
    "macro": true,
    "col": 1,
    "name-col": 2,
    "context": {},
    "end-col": 21,
    "arity": 2,
    "varargs-min-arity": 1,
    "row": 5,
    "to": "clojure.test"
  }
]

 ```

**Comment by ericdallo** [comment]
oh, actually it's correct, that testing is from the require, let me double-check why it didn't work with clojure-lsp

**Comment by borkdude** [comment]
@ericdallo I certainly broke something 😭  but I'll fix it 😄

**Comment by ericdallo** [comment]
@borkdude my bad, I forgot to use the local clj-kondo version on local clojure-lsp 🤦🏻‍♂️ 
It works perfectly now!
image

**Comment by borkdude** [comment]
Nonetheless I had to fix a NPE. Good!

File: src/clj_kondo/impl/analyzer/test.clj (hunk 1 above)
Comment (ericdallo) [reply src/clj_kondo/impl/analyzer/test.clj:47]: I'm ok with that

**Pull request merged**